pkg/kubeapiserver/resource_handler.go

```diff
@@ -49,6 +49,19 @@ func (r *ResourceHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	gvr := schema.GroupVersionResource{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion, Resource: requestInfo.Resource}
 
 	clusterName := request.ClusterNameValue(req.Context())
+	// When clusterName not empty, first check cluster whether exist
+	if clusterName != "" {
+		if _, err := r.clusterLister.Get(clusterName); err != nil {
+			klog.ErrorS(err, "Failed to handle resource request, not get cluster from cache", "cluster", clusterName, "resource", gvr)
+			err = fmt.Errorf("not found request cluster")
+			responsewriters.ErrorNegotiated(
+				apierrors.NewInternalError(err),
```

Comment:
If the error is cluster not found, then instead of returning InternalError, we return BadRequest(NewBadRequest) and do not need to print the log.

Reply:
done